src/ensemble/EnsembleBase.h

```diff
@@ -8,11 +8,10 @@
 
 #include "DomainBase.h"
 #include "molecules/MoleculeForwardDeclaration.h"
+#include "molecules/mixingrules/MixingRuleBase.h"
 
 class ParticleContainer;
 
-class MixingRuleBase;
```

Comment:
@FG-TUM 
Out of interest and since I didn't know what to google:
What is the purpose of this `class MixingRuleBase` and other classes here? What is the difference to `#include MixingRuleBase.h`?
Thank you :)

Reply:
It's called "forward declaration".

Very short explanation:
You tell the compiler "hey the class `MixingRuleBase` exists but no need to worry about any details (= memory layout, definitions, ...). I guarantee that as soon as you'll need this info (= during linking) it will be there (because the definitions will end up in the same compilation unit via other includes)". This technique is needed if the include would create a circular include.

Long but better explanation as always on [StackOverflow](https://stackoverflow.com/questions/4757565/what-are-forward-declarations-in-c).